tests/src/test/java/com/maxio/advancedbilling/controllers/events/EventsControllerListEventsTest.java

```diff
@@ -0,0 +1,313 @@
+        assertThat(subscriptionProductChange.getNewProductId()).isEqualTo(product2.getId());
+
+        event = eventsMap.get(SUBSCRIPTION_STATE_CHANGE);
+        SubscriptionStateChange subscriptionStateChange = event.getEventSpecificData().match(new EventSpecificDataGetter<>());
+        assertThat(event.getMessage()).isEqualTo("State changed on %s's subscription to %s from active to canceled"
+                .formatted(updatedCustomerFullName, product2.getName()));
+        assertThat(event.getSubscriptionId()).isEqualTo(subscription.getId());
+        assertThat(subscriptionStateChange.getPreviousSubscriptionState()).isEqualTo("active");
+        assertThat(subscriptionStateChange.getNewSubscriptionState()).isEqualTo("canceled");
+
+        event = eventsMap.get(SUBSCRIPTION_DELETION);
+        assertThat(event.getMessage()).startsWith("Subscription for %s".formatted(updatedCustomerFullName))
+                .endsWith("/%s) was deleted".formatted(updatedCustomer.getEmail()));
+        assertThat(event.getSubscriptionId()).isEqualTo(subscription.getId());
+        assertThat(event.getEventSpecificData()).isNull();
+
+        event = eventsMap.get(ITEM_PRICE_POINT_CHANGED);
+        ItemPricePointChanged itemPricePointChanged = event.getEventSpecificData().match(new EventSpecificDataGetter<>());
+        assertThat(event.getMessage()).isEqualTo("Product price point has been changed");
+        assertThat(event.getSubscriptionId()).isEqualTo(subscription.getId());
+        assertThat(itemPricePointChanged.getItemType()).isEqualTo("Product");
+        assertThat(itemPricePointChanged.getItemId()).isNotNull();
+        assertThat(itemPricePointChanged.getItemName()).isNotNull();
+        assertThat(itemPricePointChanged.getItemHandle()).isNotNull();
+        assertThat(itemPricePointChanged.getPreviousPricePoint()).isNotNull();
+        assertThat(itemPricePointChanged.getCurrentPricePoint()).isNotNull();
+    }
+
+    @Test
+    void shouldListEventsWithFilter() throws IOException, ApiException {
+        List<EventResponse> events = client.getEventsController().listEvents(new ListEventsInput.Builder()
+                .sinceId(lastEventId)
+                .direction(Direction.DESC)
+                .filter(List.of(PAYMENT_SUCCESS, METERED_USAGE))
+                .build()
+        );
+        assertThat(events).hasSize(2);
+
+        Map<EventKey, Event> eventsMap = events.stream()
+                .collect(Collectors.toMap(e -> e.getEvent().getKey(), EventResponse::getEvent));
+
+        // then
+        for (EventResponse event : events) {
+            assertCommonFields(event.getEvent());
+        }
+
+        assertThat(eventsMap.keySet()).containsExactlyInAnyOrder(PAYMENT_SUCCESS, METERED_USAGE);
+
+        Event event = eventsMap.get(PAYMENT_SUCCESS);
+        assertThat(event.getMessage()).isEqualTo("Successful payment of $10.00 for %s's subscription to %s"
```

Comment:
Assertions for `PAYMENT_SUCCESS` and `METERED_USAGE` could be extracted and reused.